### PR TITLE
Stats for event count per contact and per location

### PR DESCRIPTION
**Pull request by gzagatti**
### I have:

- [x] Formatted any Python files with [black](https://github.com/ambv/black)
- [x] Brought the branch up to date with master
- [x] Added any relevant Github labels
- [x] Added tests for any new additions
- [x] Added or updated any relevant documentation
- [ ] Added an Architectural Decision Record (ADR), if appropriate
- [x] Added an [MPLv2 License Header](https://www.mozilla.org/en-US/MPL/headers/) if appropriate
- [ ] Updated the [Changelog](https://github.com/Flowminder/FlowKit/blob/master/CHANGELOG.md) 

### Description

This PR implements two covariate classes `PerLocationEventCount` and `PerContactEventCount` which calculate statistics for event counts sliced per location or per contact for each subscriber.

### CODE REVIEW

File: flowmachine/flowmachine/features/subscriber/event_count.py [hunk 1]
```diff
@@ -99,8 +102,267 @@ def _make_query(self):
...
+        self.statistic = statistic
+
+        if self.statistic not in valid_stats:
+            raise ValueError(
+                "{} is not a valid statistic. Use one of {}".format(
+                    self.statistic, valid_stats
+                )
+            )
+
+        self.contact_balance = ContactBalance(
```
Comment (greenape) [inline flowmachine/flowmachine/features/subscriber/event_count.py]: Bubble up?

**Comment by gzagatti** [comment]
Checks are failling due to CircleCI issue, it seems like `Partially Degraded Service - components affected: Docker Hub`

File: flowmachine/flowmachine/features/subscriber/event_count.py [hunk 2]
```diff
@@ -99,8 +102,258 @@ def _make_query(self):
...
+    """
+
+    def __init__(
+        self,
+        start,
+        stop,
+        contact_balance,
+        statistic="avg",
+        *,
+        hours="all",
```
Comment (greenape) [inline flowmachine/flowmachine/features/subscriber/event_count.py]: Can we get rid of these please - they now don't do anything.

Comment (gzagatti) [reply flowmachine/flowmachine/features/subscriber/event_count.py]: Of course!

File: CHANGELOG.md [hunk 3]
```diff
@@ -18,7 +18,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 - Dockerised development setup, with support for live reload of `flowmachine` and `flowapi` after source code changes.
 - Pre-commit hook for Python formatting with black.
-- Added new `IntereventPeriod`, `ContactReciprocal`, `ProportionContactReciprocal`, `ProportionEventReciprocal`, `ProportionEventType`, `MDSVolume` query classes to FlowMachine.
+- Added new `IntereventPeriod`, `ContactReciprocal`, `ProportionContactReciprocal`, `ProportionEventReciprocal`, `ProportionEventType`, `MDSVolume`, `PerLocationEventCount`, `PerContactEventCount` query classes to FlowMachine.
```
Comment (greenape) [inline CHANGELOG.md]: These should be in the Unreleased section

**Pull request merged**